sdk/go/http.go

```diff
@@ -34,17 +38,23 @@ var (
 	defCertFile = fmt.Sprintf("%s%s%s", os.Getenv("GOPATH"), defCertsPath, "mainflux-server.crt")
 	defKeyFile  = fmt.Sprintf("%s%s%s", os.Getenv("GOPATH"), defCertsPath, "mainflux-server.key")
 	defCaFile   = fmt.Sprintf("%s%s%s", os.Getenv("GOPATH"), defCertsPath, "ca.crt")
+
+	limit  = 10
+	offset = 0
+
+	isHTTPS = false
 )
 
 // SetServerAddr - set addr using host and port
-func SetServerAddr(proto string, host string, port int) {
-	serverAddr = fmt.Sprintf("%s://%s", proto, host)
-
-	if port != 0 {
-		serverAddr = fmt.Sprintf("%s:%s", serverAddr, strconv.Itoa(port))
+func SetServerAddr(proto, host, port string) {
+	if proto == "https" {
+		isHTTPS = true
 	}
+	serverAddr = fmt.Sprintf("%s://%s:%s", proto, host, port)
```

Comment:
Maybe you should create server struct and add addr field to it. Also, you could add `New` for this server struct and pass these parameters.